components/datetime/src/format/datetime.rs

```diff
@@ -284,10 +284,26 @@ where
                     .month()
                     .ok_or(Error::MissingInputField(Some("month")))?
                     .code;
+
                 let symbols = date_symbols
                     .ok_or(Error::MissingDateSymbols)?
                     .get_symbols_for_month(month, length, code)?;
-                let symbol = symbols.get(code).ok_or(Error::MissingMonthSymbol(code))?;
+
+                let symbol_option = symbols.get(code); //.ok_or(Error::MissingMonthSymbol(code))?;
+                let symbol = if symbol_option.is_none() {
```

Comment:
I would use `if let Some(symbol) = symbol_option {} else {}`

I would also put the `write_str` call inside the if let, so no need to `let symbol = `.  This way the else branch can pick which order to print things in (and we probably want to output the `(leap)` _after_ `symbol`)

Reply:
I put `write_str` directly inside the if statement, so there is no more need for `symbol` or an `if let`